### PR TITLE
feat(layout:sider): support for custom trigger with nzCollapsedWidth set to 0

### LINKED ISSUE #1950: support custom trigger for 0 width collapsed sider
## What problem does this feature solve?
I'm trying to build an application with two siderbars, one for navigation, and the other one for a rapid access to some important informations. The first one is working great and has a nice sider trigger using the default one. But the second one should use a custom trigger template as I don't want the same icon. because [nzTrigger] only affects the trigger at the bottom of the page and not the one that shows when you have a 0px collapsed sider, I have no way to change it right now.
## What does the proposed API look like?
simply have [nzTrigger] affect the collapsed trigger too, or have a [nzZeroWidthTrigger] property that takes a template too, in order to provide a custom icon for this trigger. Could also just be an icon string.

### PR DESCRIPTION
**Pull request by Supamiu**
closes #1950

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] (*only english docs*) Docs have been added / updated (for bug fixes / features) 

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1950 

## What is the new behavior?

You can now set a custom trigger tof rth trigger shown when you have a nzCollapsedWidth set to 0 on 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by Supamiu** [comment]
@vthinkxie I tried to add some tests but couldn't make them work properly as I'm not very familiar with your testing environment.

I pushed the current state of my tests, hoping you can fix them.

File: components/layout/nz-layout.spec.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ import { NzLayoutComponent } from './nz-layout.component';
 import { NzLayoutModule } from './nz-layout.module';
 import { NzSiderComponent } from './nz-sider.component';
 
-describe('layout', () => {
+fdescribe('layout', () => {
```
Comment (vthinkxie) [inline components/layout/nz-layout.spec.ts]: should not use fdescribe here

**Comment by vthinkxie** [comment]
Hi @Supamiu would you plz rebase this pr onto master, thanks.

**Comment by Supamiu** [comment]
Hi, I'm really sorry I didn't get time to finish the tests, couldn't make them work with the custom sider icon. Do you have a guide or something about your unit tests so I could check how to implement them properly? 

Or if you have a bit of time, I could use a small review on them just in case you see what I did wrong.

**Comment by vthinkxie** [comment]
don't worry about this, I just take it over, thanks for your pr!

**Pull request closed without merging**